corehq/apps/es/apps.py

```diff
@@ -23,7 +24,10 @@ def builtin_filters(self):
         ] + super(AppES, self).builtin_filters
 
     def build_comment(self, comment):
-        return self.search_string_query(comment, default_fields=['build_comment'])
+        return self.add_query(queries.search_string_query(comment, ['build_comment']), queries.SHOULD)
+
+    def version(self, version):
+        return self.add_query(filters.term('version', version), queries.SHOULD)
```

Comment:
I'm not crazy about this approach - I'd expect `build_comment` to filter the results to that comment and `version` to filter results to that version.  That is, `app_es.build_comment(q).version(q)` would AND the queries together, but this `OR`s them.  What about building instead for this client code?

```python
app_es.set_query(build_comment(query), queries.SHOULD)
app_es.set_query(version(query), queries.SHOULD)
```

That would ensure that the client code knows how the queries are being joined together.

Reply:
Like 229cab8388f? My head isn't totally wrapped around the ES logic, but I do agree that chaining functions seems like and AND operation.

I moved `build_comment` and `version` out of `AppES` to be consistent with the rest of the file, but I'd rather make them class methods. I like that the helper methods are all short names, but it's a little weird to import a method named `version` so it's now available throughout this particular file, which is likely to use the string "version" as a variable name. (There aren't actually any name conflicts, but I can imagine them being added.) What do you think?

Reply:
You could make a `search_comment_and_version` method on the `AppES` class if you prefer that framing (or if you think that concept is reusable).  That's an excellent point about importing - I'd use namespaces for that (I think I've done similar elsewhere)
```python
from corehq.apps.es import apps as app_es

...
        app_es.version(...)
```
That actually conflicts with the query, which is here called `app_es` (I find that name misleading, FWIW - maybe `app_query` or something), but you get the idea.

Reply:
To add to that - `version` and `build_comment` are perfectly suitable names when viewed _in the context of_ apps elasticsearch queries, and I'd prefer to bring along that namespace via the method I described above versus using a convention-based namespace like `app_es_version` and `app_es_build_comment`